bin/annotate_test_failures

```diff
@@ -100,6 +115,63 @@ def update_annotation(title, list, style, state)
   end
 end
 
+# Given a list of failures, parse list and send a slack notification with the test names in the payload
+#
+def send_slack_notification(slack_webhook, slack_channel, title, list)
+    failing_tests = list.map { |item| "#{item.name} in #{item.classname}" }        
+    assertion_failures_count = list.count
+    test_text = (assertion_failures_count == 1) ? "Test" : "Tests"
+
+    slack_message_payload = {
+      "channel": "#{slack_channel}",
+      "username": "#{ENV['BUILDKITE_PIPELINE_NAME']} Tests Failures",
+      "icon_emoji": ":fire:",
+      "blocks": [
+        {
+          "type": "section",
+          "text": {
+            "type": "mrkdwn",
+            "text": ":warning: *#{assertion_failures_count} #{test_text} Failed in #{ENV['BUILDKITE_PIPELINE_NAME']} - #{title}*"
+          },
+          "accessory": {
+            "type": "button",
+            "text": {
+              "type": "plain_text",
+              "text": "Build",
+              "emoji": true
+            },
+            "value": "build",
+            "url": "https://buildkite.com/#{ENV['BUILDKITE_ORGANIZATION_SLUG']}/#{ENV['BUILDKITE_PIPELINE_SLUG']}/builds/#{ENV['BUILDKITE_BUILD_NUMBER']}##{ENV['BUILDKITE_JOB_ID']}",
+            "action_id": "button-action"
+          }
+        },
+        {
+          "type": "divider"
+        },
+        {
+          "type": "section",
+          "text": {
+            "type": "mrkdwn",
+            "text": "*Failing #{test_text}:* #{failing_tests.to_s[1..-2]}\n" # To remove the bracket from the notification and only list the tests
```

Comment:
Instead of using `.to_s` to convert the array into its debug representation (that will contain the brackets), I'd suggest you to use `.join(',')` to join the items of the array into a nice-looking list.

You could even decide to join the list of failing tests using `.join("\n")` instead if you prefer them to be rendered each on their own line, In which case I'd advocate to also add `-` in front of each entry to make it look like a bullet-point list in Slack 🙂 And while we're at prettifying all this, you can also use backticks around the test and class name too 🙃 

```ruby
failing_tests = list.map { |item| " - `#{item.name}` in `#{item.classname}`" }
```
```ruby
   "text": "*Failing #{test_text}:*\n#{failing_tests.join("\n")}"
```

Reply:
thanks for the re-review! 

agreed and made all the changes in https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/60/commits/32ef08fe69d5fd0c2f5b0de90ddac3fcd0c08856 except for displaying them as a bullet point list. found that Slack message, when sent by the block API, only covers half the screen ([looks like a known issue](https://stackoverflow.com/questions/57438097/how-to-make-slack-api-block-kit-take-up-the-entire-width-of-the-slack-window)) and because some test names can be pretty long, for those, the message doesn't look that great (screenshot below from a local test, which is why the default label "Tests" is displayed):

<img width="680" alt="image" src="https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/assets/17252150/680a3f6f-2897-4f1b-8f92-2934f6fbefc0">